Documentation/app-operations.md

```diff
@@ -0,0 +1,225 @@
+
+```bash
+● mysql.service - Application=mysql Image=registry-1.docker.io/library/mysql
+   Loaded: loaded (/usr/lib64/systemd/system/mysql.service; static; vendor preset: enabled)
+      Active: active (running) since Wed 2016-07-09 17:00:50 UTC; 8min 38s ago
+       Main PID: 6 (mysqld)
+          CGroup: /machine.slice/machine-rkt\x2d11efeefb\x2d2e8d\x2d47d0\x2da4ee\x2de1631327acac.scope/system.slice/mysql.service
+                     └─6 mysqld
+
+  Jul 15 17:01:03 rkt-$uuid mysql[6]: [Note]   - '::' resolves to '::';
+  Jul 13 17:01:03 rkt-$uuid mysql[6]: [Note] Server socket created on IP: '::'.
+  Jul 13 17:01:03 rkt-$uuid mysql[6]: [Note] InnoDB: Loading buffer pool(s) from /var/lib/mysql/ib_buffer_pool
+  Jul 13 17:01:03 rkt-$uuid mysql[6]: [Warning] 'db' entry 'sys mysql.sys@localhost' ignored in --skip-name-resolve mode.
+  Jul 13 17:01:03 rkt-$uuid mysql[6]: [Warning] 'proxies_priv' entry '@ root@localhost' ignored in --skip-name-resolve mode.
+  Jul 13 17:01:03 rkt-$uuid mysql[6]: [Note] InnoDB: Buffer pool(s) load completed at 160713 17:01:03
+  Jul 13 17:01:03 rkt-$uuid mysql[6]: [Warning] 'tables_priv' entry 'sys_config mysql.sys@localhost' ignored in --skip-name-resolve mode.
+  Jul 13 17:01:03 rkt-$uuid mysql[6]: [Note] Event Scheduler: Loaded 0 events
+  Jul 13 17:01:03 rkt-$uuid mysql[6]: [Note] mysqld: ready for connections.
+  Jul 13 17:01:03 rkt-$uuid mysql[6]: Version: '5.7.13'  socket: '/var/run/mysqld/mysqld.sock'  port: 3306  MySQL Community Server (GPL)
+```
+
+## rkt app exec
+Executes a command inside an application.
+
+```bash
+rkt app exec <pod-uuid> --app=<app-name> <arguments> -- <command> <command-arguments>
+```
+
+# Entrypoints
+
+In order to facilitate the app-level operations API, 4 new entrypoints are introduced.
+These entrypoints are stage1 binaries that are capable of communicating with the stage1 supervisor.
+
+For example, the `app start` entrypoint in default stage1 flavor sends a `start` signal
+(similar to `systemctl start`) for the service files of the application to the systemd which is the stage1 supervisor.
+
+## rkt app inject
+
+`coreos.com/rkt/stage1/app/inject`
+
+1. receive the prepared-app intended for injection consisting of it's rootfs, unit files, volumes, etc.
+2. `coreos.com/rkt/stage1/inject` entrypoint is resolved via annotations found within `/var/lib/rkt/pods/run/$uuid/stage1/manifest`.
+3. the entrypoint will be executed and passed a reference to the rootfs of the application that was prepared.
+
+The responsibility of this entrypoint is to receive a prepared app, copy its contents into stage2, 
+so that it is ready to be started by the stage1 supervisor (e.g. systemd for the default flavor).
+
+### Stage1 Default Flavor Example Workflow
+
+1. Configure the application service files given an application image/images.
+2. Copy the application rootfs to `/var/lib/rkt/pods/run/$uuid/stage1/rootfs/opt/stage2/`.
```

Comment:
This should be fixed since it's done by stage0 in two steps: prepare and run. It's not copying and in the current implementation depends ig using overlayfs or not (see comments in the previous commit).